pilot/pkg/model/service.go

```diff
@@ -639,6 +639,55 @@ func (h Hostnames) Swap(i, j int) {
 	h[i], h[j] = h[j], h[i]
 }
 
+// Intersection returns the subset of host names that are covered by both h and other.
+// e.g.:
+//  Hostnames(["foo.com","bar.com"]).Intersection(Hostnames(["*.com"]))         = Hostnames(["foo.com","bar.com"])
+//  Hostnames(["foo.com","*.net"]).Intersection(Hostnames(["*.com","bar.net"])) = Hostnames(["foo.com","bar.net"])
+//  Hostnames(["foo.com","*.net"]).Intersection(Hostnames(["*.bar.net"]))       = Hostnames(["*.bar.net"])
+//  Hostnames(["foo.com"]).Intersection(Hostnames(["bar.com"]))                 = Hostnames([])
+func (h Hostnames) Intersection(other Hostnames) Hostnames {
+	result := make(Hostnames, 0, len(h))
+	for _, hHost := range h {
+		for _, oHost := range other {
+			if hHost.SubsetOf(oHost) {
+				result = append(result, hHost)
+			} else if oHost.SubsetOf(hHost) {
+				result = append(result, oHost)
+			}
+		}
+	}
+	return result
+}
```

Comment:
could you add a test for this? that will be the ultimate validation and prevent people from messing this logic up